hibernate-core/src/main/java/org/hibernate/Hibernate.java

```diff
@@ -187,7 +202,10 @@ public static boolean isPropertyInitialized(Object proxy, String propertyName) {
 
 		if ( entity instanceof PersistentAttributeInterceptable ) {
 			PersistentAttributeInterceptor interceptor = ( (PersistentAttributeInterceptable) entity ).$$_hibernate_getInterceptor();
-			if ( interceptor != null && interceptor instanceof LazyAttributeLoadingInterceptor ) {
+			if ( interceptor instanceof EnhancementAsProxyLazinessInterceptor ) {
+				return false;
+			}
+			if ( interceptor instanceof LazyAttributeLoadingInterceptor ) {
 				return ( (LazyAttributeLoadingInterceptor) interceptor ).isAttributeLoaded( propertyName );
```

Comment:
Currently, ```Hibernate.isPropertyInitialized( activity, "instruction" )``` will return ```true``` if the attribute is initialized to an uninitialized proxy.

At the same time, ```Hibernate.isInitialized( activity.getInstruction() )``` will return ```false```.

I find this confusing. IMO, the return values for both should be ```false```.

Reply:
I see that in ```FetchGraphTest#testLoadNonOwningOneToOne```, that this is intended.

It would be helpful to add documentation to this method that a return value of ```true``` could indicate that the property can be initialized to an uninitialized collection or proxy.

Reply:
I disagree.  The contract of isPropertyInitialized is whether that value is available on the class

Reply:
Unless I am off base, the same should be true in 4.x if an object is enhanced and has a proxy for an association.